src/spec/template_spec.js

```diff
@@ -0,0 +1,279 @@
+  });
+
+  it('should show the save template button in edit mode in english', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+
+    var templatePage = new TemplatePage();
+
+    // Act
+    var saveTemplateButtonIsDisplayed = templatePage.isSaveButtonDisplayed();
+
+    // Assert
+    expect(saveTemplateButtonIsDisplayed).toBeTruthy();
+  });
+
+  it('should not allow to save template with empty body content', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.cleanTemplateBodySection();
+    templatePage.clickSaveTemplateButton();
+
+    // Assert
+    expect(templatePage.isInvalidTemplateBody()).toBeTruthy();
+  });
+
+  it('should save template in edit mode and redirect to templates list', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+    var body = "Test content body";
+
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.FillTemplateBodySection(body);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(browser.getCurrentUrl()).not.toContain('templates/');
+  });
+
+  it('should save template in edit mode and redirect to templates list', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+    var body = "Test content body";
+
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.FillTemplateBodySection(body);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(browser.getCurrentUrl()).not.toContain('templates/');
+  });
+
+  it('should create a new template', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/new');
+    setupTemplateResponse();
+    var templateName = "Test template";
+    var templateFromEmail = "test@test.test";
+    var templateFromName = "test name";
+    var templateSubject = "test subject";
+    var templateBody = "Template body test"
+    var expectedGuid = "ID: 1169ab26-78dc-4474-a8c2-e1f02f6c05ed";
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.fillTemplateNameInput(templateName);
+    templatePage.fillTemplateFromEmailInput(templateFromEmail);
+    templatePage.fillTemplateFromNameinput(templateFromName);
+    templatePage.fillTemplateSubjectInput(templateSubject);
+    templatePage.FillTemplateBodySection(templateBody);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(browser.getCurrentUrl()).not.toContain('templates/new');
+    expect(templatePage.getLastestSquareGuidCreatedText()).toBe(expectedGuid);
```

Comment:
> _should create a new template_

What does it mean? I did not see that a new template is created in these assertions.